0000-frontend-development-infrastructure-and-workflow-changes.md

```diff
@@ -0,0 +1,75 @@
+- Feature Name: frontend-development-infrastructure-and-workflow-changes
+- Start Date: 2016-07-20
+- RFC PR: 
+- Issue: 
+
+# Summary
+[summary]: #summary
+
+Currently, client side javascript in foreman is implemented with native javascript and the jquery library and managed by the Rails asset pipeline. In addition, various unrelated libraries are used for specific tasks, charting for example. This RFC proposes adding the React javascript library to the technology stack via NPM and Webpack. [(see Webpack PR)](https://github.com/theforeman/foreman/pull/3433) 
```

Comment:
Is NPM/Webpack required in order to implement React?

We usually prefer evolution patches to big bangs. Authors are asked to break huge PRs into smaller pieces anyway.